katharsis-client/src/main/java/io/katharsis/client/internal/DefaultQueryParamsSerializer.java

```diff
@@ -0,0 +1,119 @@
+		Map<String, Set<String>> map = new HashMap<String, Set<String>>();
+		TypedParams<FilterParams> filters = queryParams.getFilters();
+		if (filters != null) {
+			Map<String, FilterParams> filterParamsMap = filters.getParams();
+			for (Map.Entry<String, FilterParams> entry : filterParamsMap.entrySet()) {
+				String type = entry.getKey();
+				Map<String, Set<String>> params = entry.getValue().getParams();
+				for (Map.Entry<String, Set<String>> param : params.entrySet()) {
+					String paramName = RestrictedQueryParamsMembers.filter.name() + "[" + type + "]" + serializeKey(param.getKey());
+					map.put(paramName, param.getValue());
+				}
+			}
+		}
+		return map;
+	}
+
+	private String serializeKey(String key) {
+		return "[" + key.replace(".", "][") + "]";
+	}
+
+	@Override
+	public Map<String, String> serializeSorting(QueryParams queryParams) {
+		Map<String, String> map = new HashMap<String, String>();
+		TypedParams<SortingParams> sorting = queryParams.getSorting();
+		if (sorting != null) {
+			Map<String, SortingParams> sortingParamsMap = sorting.getParams();
+			for (Map.Entry<String, SortingParams> entry : sortingParamsMap.entrySet()) {
+				String type = entry.getKey();
+				Map<String, RestrictedSortingValues> params = entry.getValue().getParams();
+				for (Entry<String, RestrictedSortingValues> param : params.entrySet()) {
+					String paramName = RestrictedQueryParamsMembers.sort.name() + "[" + type + "][" + param.getKey() + "]";
+					map.put(paramName, param.getValue().toString());
+				}
+			}
+		}
+		return map;
+	}
+
+	@Override
+	public Map<String, Set<String>> serializeIncludedFields(QueryParams queryParams) {
+		Map<String, Set<String>> map = new HashMap<String, Set<String>>();
+		TypedParams<IncludedFieldsParams> includedFields = queryParams.getIncludedFields();
+		if (includedFields != null) {
+			Map<String, IncludedFieldsParams> includedFieldsParamsMap = includedFields.getParams();
+			for (Entry<String, IncludedFieldsParams> entry : includedFieldsParamsMap.entrySet()) {
+				String type = entry.getKey();
+				Set<String> values = entry.getValue().getParams();
+				String paramName = RestrictedQueryParamsMembers.fields.name() + "[" + type + "]";
+				map.put(paramName, values);
+			}
+		}
+		return map;
+	}
+
+	@Override
+	public Map<String, Set<String>> serializeIncludedRelations(QueryParams queryParams) {
+		Map<String, Set<String>> map = new HashMap<String, Set<String>>();
+		TypedParams<IncludedRelationsParams> includedRelations = queryParams.getIncludedRelations();
+		if (includedRelations != null) {
+			Map<String, IncludedRelationsParams> includedRelationsParamsMap = includedRelations.getParams();
+			for (Entry<String, IncludedRelationsParams> entry : includedRelationsParamsMap.entrySet()) {
+				String type = entry.getKey();
+				Set<Inclusion> inclusions = entry.getValue().getParams();
+
+				Set<String> strInclusions = new HashSet<String>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)